src/activity/service.ts

```diff
@@ -0,0 +1,52 @@
+import { Activity, ActivityOptions } from "./activity";
+import { defaultLogger, Logger, YagnaApi } from "../utils";
+import { AgreementPoolService } from "../agreement";
+import { PaymentService } from "../payment";
+
+interface ActivityServiceOptions extends ActivityOptions {}
+
+export class ActivityPoolService {
```

Comment:
Looks like the file name has to be adjusted.

Reply:
This is very analogous to `AgreementPoolService`. This is a very primitive implementation of pool. It has a `get` and `release` method. After the task is finished, the activity is returned to the pool for reuse or terminated (if allowReuse = false)
We can change the name to `ActivityService`, but I think it implements a very simple pooling mechanism. In its current form, this service is only used by `GolemRuntime` in `golem-sdk-worker`, but it can be used (and extended) in `TaskExecutor` or `WorkContext` in the future.